VmAgent.Core/Dependencies/BasicAssetExtractor.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Azure.Gaming.VmAgent.Core.Dependencies
+{
+    public class BasicAssetExtractor
+    {
+        public static BasicAssetExtractor Instance = new BasicAssetExtractor();
+
+        private readonly ISystemOperations _systemOperations;
+        private readonly MultiLogger _logger;
+
+        public BasicAssetExtractor(ISystemOperations systemOperations = null, MultiLogger logger = null)
```

Comment:
>= null [](http://example.com/codeflow?start=70&length=6)

Should this constructor use the defaults:

new BasicAssetExtractor(SystemOperations.Default, Globals.MultiLogger)

Otherwise the problem I see is that just doing 'new BasicAssetExtractor()' with null defaults does not give you an instance that actually works (I think it will throw a NullRef error if you try calling 'ExtractAssets()').

Reply:
Thanks Pedro, I put this code **BasicAssetExtractor.Instance = new BasicAssetExtractor(SystemOperations.Default, Globals.MultiLogger)** in startup.cs to instantiate it. you think I need to update it? I also added the comment for it above ! :)